test/Microsoft.AspNet.TestHost.Tests/ClientHandlerTests.cs

```diff
@@ -234,5 +243,33 @@ public Task ExceptionBeforeFirstWriteIsReported()
             var ex = await Assert.ThrowsAsync<HttpRequestException>(() => response.Content.ReadAsStringAsync());
             Assert.IsType<InvalidOperationException>(ex.GetBaseException());
         }
+
+        private class DummyApplication : IHttpApplication<Context>
+        {
+            Func<Context, Task> _application;
+
+            public DummyApplication(Func<Context, Task> application)
```

Comment:
Is the full Context ever needed? It looks like all the tests just use HttpContext. Change this to `Func<HttpContext, Task>` / RequestDelegate?